overlord/devicestate/devicemgr.go

```diff
@@ -1815,7 +1815,16 @@ func (m *DeviceManager) SystemAndGadgetInfo(wantedSystemLabel string) (*System,
 	if err := s.LoadAssertions(nil, nil); err != nil {
 		return nil, nil, fmt.Errorf("cannot load assertions for label %q: %v", wantedSystemLabel, err)
 	}
```

Comment:
this can go as well, no? systemFromOpenSeed does LoadAssertions

Reply:
I split `systemFromSeed()` (which will do the LoadAssertions() indeed) and created a helper "systemFromOpenSeed" that assumes the seed is already open and assertion loaded. This avoids the extra LoadAssertions as systemFromOpenSeed() will not do that. However given that it confuses you (or me if I miss something) it seems like it's not the best name/split? Maybe something more explicit like `systemFromSeedWithLoadedAssertions  or something like this?

Reply:
yes, using Open to mean LoadedAssertions is confusing but then is also unclear at this point why we don't want that one to LoadAssertions given that we are loading them in the end

Reply:
basically I'm wondering if we can split out only the Open part, see other comment

Reply:
Yet another approach would be to have an helper that returns both System and seed.Seed and have the old helper just a wrapper that drops seed.Seed

Reply:
This is a great suggestion - I implemented the `loadSeedAndSystem` helper now